interfaces/cython/cantera/test/test_composite.py

```diff
@@ -718,28 +718,36 @@ def test_yaml_outunits2(self):
                              gas2.forward_rate_constants)
         self.assertArrayNear(gas.mix_diff_coeffs, gas2.mix_diff_coeffs)
 
-    def test_yaml_surface(self):
-        gas = ct.Solution('ptcombust.yaml', 'gas')
-        surf = ct.Interface('ptcombust.yaml', 'Pt_surf', [gas])
-        gas.TPY = 900, ct.one_atm, np.ones(gas.n_species)
-        surf.coverages = np.ones(surf.n_species)
-        surf.write_yaml('ptcombust-generated.yaml')
-
+    def check_ptcombust(self, gas, surf):
         generated = utilities.load_yaml("ptcombust-generated.yaml")
-        for key in ('phases', 'species', 'gas-reactions', 'Pt_surf-reactions'):
+        for key in ("phases", "species", "gas-reactions", "Pt_surf-reactions"):
             self.assertIn(key, generated)
-        self.assertEqual(len(generated['gas-reactions']), gas.n_reactions)
-        self.assertEqual(len(generated['Pt_surf-reactions']), surf.n_reactions)
-        self.assertEqual(len(generated['species']), surf.n_total_species)
+        self.assertEqual(len(generated["gas-reactions"]), gas.n_reactions)
+        self.assertEqual(len(generated["Pt_surf-reactions"]), surf.n_reactions)
+        self.assertEqual(len(generated["species"]), surf.n_total_species)
 
-        gas2 = ct.Solution('ptcombust-generated.yaml', 'gas')
-        surf2 = ct.Solution('ptcombust-generated.yaml', 'Pt_surf', [gas2])
+        surf2 = ct.Solution("ptcombust-generated.yaml", "Pt_surf")
         self.assertArrayNear(surf.concentrations, surf2.concentrations)
         self.assertArrayNear(surf.partial_molar_enthalpies,
                              surf2.partial_molar_enthalpies)
         self.assertArrayNear(surf.forward_rate_constants,
                              surf2.forward_rate_constants)
 
+    def test_yaml_surface_explicit(self):
+        gas = ct.Solution('ptcombust.yaml', 'gas')
+        surf = ct.Interface('ptcombust.yaml', 'Pt_surf', [gas])
+        gas.TPY = 900, ct.one_atm, np.ones(gas.n_species)
+        surf.coverages = np.ones(surf.n_species)
+        surf.write_yaml('ptcombust-generated.yaml')
+        self.check_ptcombust(gas, surf)
+
+    def test_yaml_surface_adjacent(self):
+        surf = ct.Interface('ptcombust.yaml', 'Pt_surf')
+        gas = surf.adjacent["gas"]
+        gas.TPY = 900, ct.one_atm, np.ones(gas.n_species)
+        surf.coverages = np.ones(surf.n_species)
+        surf.write_yaml("ptcombust-generated.yaml")
```

Comment:
is there a `self.check_ptcombust(gas, surf)` missing?

Reply:
Good catch.